src/config/file_info.rs

```diff
@@ -165,12 +184,14 @@ impl FileInfo<'_, '_, '_, '_> {
         parent: P,
         idx: usize,
     ) -> Result<Vec<(PathBuf, rpm::FileOptions)>, ConfigError> {
-        self.generate_expanded_path(build_target, parent, idx)
-            .map(|p| {
-                p.iter()
-                    .map(|(src, dst)| (src.clone(), self.generate_rpm_file_options(dst)))
-                    .collect::<Vec<_>>()
-            })
+        match self.generate_expanded_path(build_target, parent, idx) {
+            Ok(v) => v.into_iter().try_fold(Vec::new(), |mut acc, (src, dst)| {
+                let rpm_file_option = self.generate_rpm_file_options(dst)?;
+                acc.push((src, rpm_file_option));
+                Ok(acc)
+            }),
+            Err(e) => Err(e),
+        }
```

Comment:
I believe it could be written concisely.
```rust
        self.generate_expanded_path(build_target, parent, idx)?
            .iter()
            .map(|(src, dst)| self.generate_rpm_file_options(dst, idx).map(|v| (src.clone(), v)))
            .collect::<Result<Vec<_>, _>>()
```
Note that 2nd argument `idx` of `generate_rpm_file_options()` shall be added to handle my previous comment.

Reply:
That is a much improved implementation, thank you for the suggestion.